lib/fund.js

```diff
@@ -37,161 +32,174 @@ fundCmd.completion = function (opts, cb) {
+        !Object.hasOwnProperty.call(prev.dependencies, node.name)) {
+        prev = node
+        parent = nodeResult
       }
-
-      res.push(result)
-
-      return res
-    }, [])
-
-  const [ result ] = getFundingItems({
-    [fundingInfo.name]: {
-      dependencies: fundingInfo.dependencies,
-      funding: fundingInfo.funding,
-      version: fundingInfo.version
+      return Object.keys(node.dependencies || {})
+        .map(key => ({
+          name: key,
+          ...node.dependencies[key]
+        }))
     }
   })
 
-  return archy(result, '', { unicode: opts.unicode })
+  const [root] = result.nodes
+  return tree(root.label !== header ? { label: header, nodes: [root] } : root)
 }
 
-function openFundingUrl (packageName, cb) {
-  function getUrlAndOpen (packageMetadata) {
-    const { funding } = packageMetadata
-    const { type, url } = retrieveFunding(funding) || {}
-    const noFundingError =
-      new Error(`No funding method available for: ${packageName}`)
-    noFundingError.code = 'ENOFUND'
-    const typePrefix = type ? `${type} funding` : 'Funding'
-    const msg = `${typePrefix} available at the following URL`
-
-    if (validFundingUrl(funding)) {
-      openUrl(url, msg, cb)
+function openFundingUrl ({ actualTree, spec, fundingSourceNumber }) {
+  const retrievePackageMetadata = () => {
+    const arg = npa(spec, actualTree.path)
+    let packageMetadata
+    if (arg.type === 'directory') {
+      if (actualTree.path === arg.where) {
+        packageMetadata = actualTree.package
+      } else {
+        for (const item of actualTree.inventory.values()) {
+          if (item.path === arg.where) {
+            packageMetadata = item.package
+          }
+        }
+      }
     } else {
-      throw noFundingError
+      const [ item ] = [
+        ...actualTree.inventory.values()
+      ]
+        .filter(i => i && i.package &&
+          i.package.name === arg.name &&
+          semver.valid(i.package.version))
+        .sort((a, b) => semver.rcompare(a.package.version, b.package.version))
+
+      if (item) {
+        packageMetadata = item.package
+      } else {
+        // TODO: reenable pacote.manifest call for non-project-dependencies
+      }
     }
+    return packageMetadata
   }
 
-  fetchPackageMetadata(
-    packageName,
-    '.',
-    { fullMetadata: true },
-    function (err, packageMetadata) {
-      if (err) return cb(err)
-      getUrlAndOpen(packageMetadata)
-    }
-  )
+  const { funding } = retrievePackageMetadata() || {}
+  const validSources = [].concat(normalizeFunding(funding)).filter(isValidFunding)
+
+  if (validSources.length === 1 || (fundingSourceNumber > 0 && fundingSourceNumber <= validSources.length)) {
+    const { type, url } = validSources[fundingSourceNumber ? fundingSourceNumber - 1 : 0]
+    const typePrefix = type ? `${type} funding` : 'Funding'
+    const msg = `${typePrefix} available at the following URL`
+    return new Promise((resolve, reject) =>
```

Comment:
Note: we should make `openUrl` return a promise.  Also, I see that it checks `process.argv` explicitly for a `--no-browser` flag, which is not great.  (Should just be checking `npm.config.get('browser') === false` or better yet, `npm.flatOptions.browser === false`.)

Reply:
created a ticket to track that here: https://github.com/npm/statusboard/issues/124